src/store/metadata/oxigraph/store.py

```diff
@@ -45,12 +46,37 @@ def setup(self):
         )
         configuration = (f"{oxigraph_url}/query", f"{oxigraph_url}/update")
         self.db = Dataset(store=SPARQLUpdateStore(*configuration))
-
+    
     def get_datasets(self) -> Optional[Dict]:  # pragma: no cover
-        """
-        Gets all datasets
-        """
-        raise NotImplementedError
+            """
+            Gets all datasets
+            """
+            graph = self.db
+
+            result: Graph = construct_datasets(graph)
+
+        # Serialize the graph into jsonld
+            data = json.loads(result.serialize(format="json-ld"))
+
+            # Use a context file to shape our jsonld, removing long form references
+            data = jsonld.flatten(
+                data, {"@context": constants.CONTEXT, "@type": ["dcat:Catalog","hydra:Collection"]}
+            )
+        
+            data["@graph"][0]["@type"] = ["dcat:Catalog","hydra:Collection"]
+            data["@graph"][0]["datasets"] = data["@graph"][0].pop("dcat:DatasetSeries")
+
+            for idx, dataset in enumerate(data["@graph"][0]["datasets"]):
```

Comment:
Mike said try not to use `enumerate` if you can achieve the same thing with a list comprehension, e.g.
```
        data["@graph"][0]["datasets"] = [
            self.get_dataset(x["@id"].split("/")[-1])
            for x in data["@graph"][0]["dcat:DatasetSeries"]
        ]
       del data["@graph"][0]["dcat:DatasetSeries"]
```